sonar-update-center-mojo/src/main/java/org/sonar/updatecenter/mojo/CompatibilityMatrix.java

```diff
@@ -70,13 +70,13 @@ public void generateHtml() throws IOException {
       Map<String, Object> dataModel = new HashMap<>();
       dataModel.put("pluginHeader", pluginModel);
 
-      CompatibilityMatrix.Plugin matrixPlugin = new CompatibilityMatrix.Plugin(plugin.getName(), plugin.getHomepageUrl(), plugin.isSupportedBySonarSource(), plugin.isBundled());
+      CompatibilityMatrix.Plugin matrixPlugin = new CompatibilityMatrix.Plugin(plugin.getName(), plugin.getHomepageUrl(), plugin.isSupportedBySonarSource());
       getPlugins().add(matrixPlugin);
 
       for (Release sq : center.getSonar().getMajorReleases()) {
         Release lastCompatible = plugin.getLastCompatible(sq.getVersion());
-        if (lastCompatible != null) {
-          matrixPlugin.getCompatibleVersionBySqVersion().put(sq.getVersion().toString(), lastCompatible.isArchived() ? "" : lastCompatible.getVersion().toString());
+        if (lastCompatible != null && !lastCompatible.isArchived()) {
```

Comment:
the condition is a bit difficult to read, a dedicated ìsLastCompatibleVersionNonArchived`would help.

And if I understand correctly, this change is actually necessary for @Zipeng's task (UPC-75).

Reply:
Fair enough, I extracted the condition to a method named `isNotArchived()`. `lastCompatible` is ensured by the variable retrieved from the previous line, it's not part of this specific condition